mountpoint-s3-crt/src/auth/credentials.rs

```diff
@@ -42,6 +54,25 @@ impl CredentialsProvider {
 
         Ok(Self { inner })
     }
+
+    /// Creates the profile credential provider.
+    pub fn new_profile(allocator: &Allocator, options: &CredentialsProviderProfileOptions) -> Result<Self, Error> {
+        auth_library_init(allocator);
+
+        // SAFETY: CredentialProvider does not modify contents of profile_name_override,
```

Comment:
Not sure I understand the safety comment: isn't the question really whether `aws_credentials_provider_new_profile` makes a deep copy of the options (and thus of `profile_name_override`)?

Reply:
There are 2 things needs to point out, to my understanding, 
* Why doe  `options.profile_name_override.as_aws_byte_cursor()` is a valid convert? by valid it means `CredentialsProvider` would not modify the contents referred by `profile_name_override`. No modified profile contents should pass in CRT.
* Once converting `profile_name_override` to `aws_byte_cursor`, `CredentialsProvider` must live at least as long as CRT as CRT might continue use the byte cursor.

Reply:
The lifetime of `CredentialsProvider` doesn't help us here -- the `options.profile_name_override` &str has lifetime `'a` which isn't tied to the `CredentialsProvider` in any way. So for this to work we need to either know that `aws_credentials_provider_new_profile` copies the string out of the byte cursor, or arrange for `CredentialsProvider` to own a copy of the string.

Reply:
CRT actually copies the bytes out of the byte cursor to a string: [aws_get_profile_nam](https://github.com/awslabs/aws-c-auth/blob/dd505b55fd46222834f35c6e54165d8cbebbfaaa/source/credentials_provider_profile.c#L377) --> [aws_string_new_from_array](https://github.com/awslabs/aws-c-sdkutils/blob/208a701fa01e99c7c8cc3dcebc8317da71362972/source/aws_profile.c#L1488) --> (aws_string_new_from_array)[https://github.com/awslabs/aws-c-common/blob/5e64b039356c72d17fcf1fb0bfc7637f8c2270f7/source/string.c#L189-L209]